api/webhook.go

```diff
@@ -550,3 +562,71 @@ func publishToQueue(queue queue.Service, p *pipeline.Build, b *library.Build, r
 		}
 	}
 }
+
+// renameRepository is a helper function that takes the old name of the repo,
+// queries the database for the repo that matches that name and org, and updates
+// that repo to its new name in order to preserve it. It also updates the secrets
+// associated with that repo.
+func renameRepository(h *library.Hook, r *library.Repo, c *gin.Context) error {
+	// get the old name of the repo
+	previousName := r.GetPreviousName()
+	// get the repo from the database that matches the old name
+	dbR, err := database.FromContext(c).GetRepo(r.GetOrg(), previousName)
+	if err != nil {
+		retErr := fmt.Errorf("%s: failed to get repo from database", baseErr)
```

Comment:
Could we enhance this error message to say what repo we couldn't find?

```suggestion
		retErr := fmt.Errorf("%s: failed to get repo %s/%s from database", baseErr, r.GetOrg(), previousName)
```